CMakeLists.txt

```diff
@@ -52,6 +52,8 @@ macro(build_uncrustify)
       -Wno-dev
     PATCH_COMMAND
       ${CMAKE_COMMAND} -E chdir <SOURCE_DIR> git checkout -q . && ${CMAKE_COMMAND} -E chdir <SOURCE_DIR> git apply -p1 --ignore-space-change --whitespace=nowarn ${CMAKE_CURRENT_SOURCE_DIR}/install-patch.diff
+    INSTALL_COMMAND
+      ${CMAKE_COMMAND} -E env DESTDIR= ${CMAKE_COMMAND} --install .
```

Comment:
I have one more question about this.  If I'm understanding all of this correctly, then if https://github.com/colcon/colcon-cmake/pull/103 goes in, we'll no longer need this workaround.  If that is the case, can you put a comment in here also describing that this is a workaround for now?  Otherwise, the solution looks fine to me.

Reply:
> if colcon/colcon-cmake#103 goes in, we'll no longer need this workaround

The RPM and deb builds don't use colcon, and in fact, they actually use `DESTDIR` instead of `CMAKE_INSTALL_PREFIX` for specifying the staging path for building the package. The other vendor packages aren't having this issue (yet), but because of this package's misbehavior regarding rebuilding the external package during the vendor package's install phase, this bug is a blocker for building RPMs or debs on a platform that doesn't already have sufficient `uncrustify` system package.

The colcon change is really just to prevent weird bugs from happening if folks invoke colcon in a weird environment.

Reply:
I see, makes sense.  In that case, I'll suggest the comment as @nuclearsandwich says, since it is non-obvious from reading the CMakeLists.txt why we need to do this.

Reply:
I added a short summary and links to upstream issues in CMake with 85d2945.